### PR TITLE
Add Tests and Linter

### PR DESCRIPTION
**Pull request by sdiao**
## Proposed change(s)

Add unit tests to the repo.
Add linter and pre-commit hook to the repo

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

https://jira.unity3d.com/browse/AIRO-673

[Failed pre-commit check example](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/actions/runs/876929225)
[Successful pre-commit check example ](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/runs/2670939891?check_suite_focus=true)

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor - Minor refactor and formatting
- [ ] Documentation update
- [x] Other (please describe) - Add unit tests

## Testing and Verification

Tested with `rosdev@a4c0aed4a281:~/ros_ws/src/ROS-TCP-Endpoint$ pytest --cov=. test/`
Screen Shot 2021-05-25 at 6 38 50 PM

### Test Configuration:
- Unity Version: N/A
- Unity machine OS + version: N/A
- ROS machine OS + version:  ROS Noetic
- ROS–Unity communication: N/A

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [x] Followed the style guidelines as described in the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the documentation as appropriate - N/A

### CODE REVIEW

File: test/test_tcp_sender.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    tcp_sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    assert tcp_sender.sender_id == 1
+    assert tcp_sender.time_between_halt_checks == 5
+    assert tcp_sender.queue == None
+    assert tcp_sender.next_srv_id == 1001
+    assert tcp_sender.srv_lock != None
+    assert tcp_sender.services_waiting == {}
+
+
+@mock.patch.object(ros_tcp_endpoint.tcp_sender.UnityTcpSender, 'send_unity_message')
```
Comment (mrpropellers) [inline test/test_tcp_sender.py:17]: Typically it's preferable to not mock our own functions unless absolutely necessary.  In this case, I _think_ you could exercise more of our code by mocking `socket.socket` instead. This also helps better validate the contract for `send_unity_error` over the implementation details. It doesn't _have_ to call `send_unity_message`, that's an implementation detail designed to eliminate redundant code, but it _does_ have to put the error message into a socket connection, so checking for `s.send` instead creates a more targeted test (in some ways, using `socket` is also an implementation detail, but at least it's lower level)

Comment (sdiao) [reply test/test_tcp_sender.py]: Make sense. I will refactor the tests.

File: test/test_tcp_sender.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+def test_send_unity_error_should_send_msg(mock_send_message):
+    sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    sender.send_unity_error("Test error")
+    mock_send_message.assert_called_once()
+
+
+@mock.patch.object(ros_tcp_endpoint.client.ClientThread, "serialize_message")
+def test_send_message_should_serialize_message(mock_serialize_msg):
+    sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    sender.send_unity_message("test topic", "test msg")
```
Comment (peifeng-unity) [inline test/test_tcp_sender.py:27]: Is it possible to assert the serialized bytes too?

Comment (sdiao) [reply test/test_tcp_sender.py:27]: We should be able to test the serialized bytes. I will look into it.

**Review by peifeng-unity — APPROVED** [review]
LGTM

**Pull request merged**